unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/UITransform/Handler/UITransformComponentHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using DCL.Controllers;
+using DCL.ECS7.UI;
+using DCL.ECSRuntime;
+using DCL.Models;
+using UnityEngine;
+using UnityEngine.UIElements;
+using Random = UnityEngine.Random;
+
+namespace DCL.ECSComponents
+{
+    public class UITransformComponentHandler : IECSComponentHandler<PBUiTransform>
+    {
+        private readonly IUIDataContainer uiDataContainer;
+        
+        public UITransformComponentHandler(IUIDataContainer dataContainer)
+        {
+            this.uiDataContainer = dataContainer;
+        }
+
+        public void OnComponentCreated(IParcelScene scene, IDCLEntity entity)
+        {
+        }
+        
+        public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
+        {
+            // If we remove an UITransform from a entity, it should have the root entity as parent
+            if(entity.parentId != SpecialEntityId.SCENE_ROOT_ENTITY)
+                entity.parentId = SpecialEntityId.SCENE_ROOT_ENTITY;
```

Comment:
What are we saving checking this condition in this case? Couldn't we directly do the assignation without the checking?

```suggestion
            entity.parentId = SpecialEntityId.SCENE_ROOT_ENTITY;
```

Reply:
That makes sense!